tests/functional/test_signals.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2018, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License with exception
+# for distributing bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+# Library imports
+# ---------------
+import signal
+
+# Third-party imports
+# -------------------
+import pytest
+from PyInstaller.utils.tests import skipif_win
+
+signals = sorted([
+    key for key in dir(signal)
+    if key.startswith('SIG') and not key.startswith('SIG_')
+])
+
+
+@skipif_win
+@pytest.mark.parametrize('signame', signals)
+@pytest.mark.parametrize('ignore', [True, False])
+def test_signal_handled(pyi_builder, signame, ignore):
+    if signame in ['SIGKILL', 'SIGSTOP']:
+        pytest.xfail('{} cannot be caught'.format(signame))
+
+    if signame in ['SIGCHLD', 'SIGCLD']:
+        pytest.xfail(
+            'Messing with {} interferes with bootloader'.format(signame)
+        )
```

Comment:
Please add a short comment. I assume these signals are not filtered out when building the `signals` list to spot whether the bootloader actually does *not* forward these to the child?!